File: GC.Domain/Gardens/Sectors/Sales/SectorSale.cs
using System;

namespace GC.Domain.Gardens.Sectors.Sales
{
    public class SectorSale
    {
        public Guid Id { get; }
        public GardenSector Sector { get; }
        public String FirstName { get; }
        public String LastName { get; }
        public String MiddleName { get; }
        public String Description { get; }
        public Int32 Price { get; }
        public String PhoneNumber { get; }
        public DateTime PublishDate { get; }

        public SectorSale(Guid id, GardenSector sector, String firstName, String lastName, String middleName, String description, Int32 price, String phoneNumber, DateTime publishDate)
        {
            Id = id;
            Sector = sector;
            FirstName = firstName;
            LastName = lastName;
            MiddleName = middleName;
            Description = description;
            Price = price;
            PhoneNumber = phoneNumber;
            PublishDate = publishDate;
        }
    }
}

File: GC.WebSpace/ClientApp/domain/gardens/sectorSales/sectorSale.ts
import { GardenSector } from "../gardenSectors/gardenSector";

export class SectorSale {
    constructor(
        public readonly id: string,
        public readonly sector: GardenSector,
        public readonly firstName: string,
        public readonly lastName: string,
        public readonly middleName: string | null,
        public readonly description: string | null,
        public readonly price: number,
        public readonly phoneNumber: string,
        public readonly publishDate: Date,
    ) { }

    public get fullName(): string {
        return `${this.lastName} ${this.firstName} ${this.middleName == null ? '' : ` ${this.middleName}`}`;
    }
}

export namespace SectorSale {
    export function toModel(data: any) {
        return new SectorSale(data.id, GardenSector.toModel(data.sector), data.firstName, data.lastName, data.middleName, data.description,
            Number(data.price), data.phoneNumber, new Date(data.publishDate));
    }

    export function toModels(data: any[]) {
        return data.map(toModel);
    }
}